src/super_gradients/training/utils/detection_utils.py

```diff
@@ -1196,4 +1233,4 @@ def compute_detection_metrics_per_cls(
     # Average over the recall_thresholds
     ap = sampled_precision_points.mean(0)
 
-    return ap, precision, recall
+    return ap, precision, recall, best_score_threshold
```

Comment:
I think it would be safer to not return the threshold here..
perhaps set it as an attribute, or return it somewhere else.